### DISCUSSION TITLE
Why is the "value" in the usePrepareContractWrite only accepts undefined?

### DISCUSSION THREAD

**Discussion by eabalderama** [wevm/wagmi#2470] [Q&A] [answered]
How do I add value to the usePrepareContractWrite hook? It is only accepting undefined. It is the same with useContractWrite. Please help. Thank you

<img width="511" alt="image" src="https://github.com/wagmi-dev/wagmi/assets/67728905/664391ee-63cd-4df4-86ad-54c224e7d534">

Edit: Although it works if I just ignore the error but during deployment the build fails because of that error.

**Comment by tmm** [comment]
What `abi` and `functionName` are you using?

**Reply by eabalderama** [reply]
Hi. Sorry for the late reply.

Here's the ABI I used
` {
    inputs: [{ internalType: 'uint256', name: '_quantity', type: 'uint256' }],
    name: 'publicMint',
    outputs: [],
    stateMutability: 'payable',
    type: 'function',
  }`

The function I used is `publicMint`

**Reply by tmm** [reply]
Works fine in this [TypeScript Playground](https://www.typescriptlang.org/play?target=99&ts=5.0.4#code/JYWwDg9gTgLgBAbzgVwM4FMAKV1gIY4DCEAdjFHgMYwDqUwM6cAvnAGZQQhwDkA7ngDmIYDwBQYyqVTwcqZABt4AXhQZsuAumJkK1Og3QAKBGLhw8AI2AAuOAG0z5xE+dxgJMMhio79pB6MUCR4CgAqAJ5g6HY8yIEATACsAGw8ADRwISAxvAD6AI7IeGQMERlwMFG5cYmpPCwAuumuztk1XpYKwJQAsoEZreYQ3l4+fs1DcDJ4jL3eVsDdVbH4EVYK6INu5lXRsWzIJNTApNvOzC3mk+aHxzCnJAByeDmryF09-WTnBIK+DgAjAkAMwkG5wABuoWQuUBJBazAAlGI5IoYGIgA).

```ts
import { usePrepareContractWrite } from 'wagmi'

const result = usePrepareContractWrite({
  abi: [
    {
      inputs: [{ internalType: 'uint256', name: '_quantity', type: 'uint256' }],
      name: 'publicMint',
      outputs: [],
      stateMutability: 'payable',
      type: 'function',
    },
  ],
  functionName: 'publicMint',
  args: [123n],
  value: 1n,
})
result
```

**Reply by jcislo** [reply]
I am having this exact same issue. Have you found a solution? It works fine locally but won't pass developer build. Not sure what to do here.

**Reply by tmm** [reply]
Can you post a minimal reproduction using a TypeScript Playground?

**Reply by eabalderama** [reply]
@jcislo I havent' found a solution yet.

**Reply by eabalderama** [reply]
So the issue appears when I import the ABI from json but when I put the specific ABI in the usePrepareContractWrite the error does not appear.

<img width="665" alt="image" src="https://github.com/wagmi-dev/wagmi/assets/67728905/c501a608-4720-4294-8e86-132ed4a5ba7a">

This is how it looks when ABI is imported from json
<img width="608" alt="image" src="https://github.com/wagmi-dev/wagmi/assets/67728905/b8ce2e68-b10f-47e9-995d-64aba3de5aad">

**Reply by jcislo** [reply]
I meant to update my message but did not get around to it. It does appear to be the JSON that is the issue and by pasting just the single function object fixed the issue for me.

**Reply by jcislo** [reply]
@eabalderama Actually even though the type error goes away in my editor and local building works... I continue to get the error on vercel deployment.

**Reply by eabalderama** [reply]
What error are you getting during the deployment?

**Reply by jcislo** [reply]
@eabalderama From vercel:


![Screenshot 2023-06-07 155513](https://github.com/wagmi-dev/wagmi/assets/55262056/c99cc300-6b6e-43bc-b1c6-60d695b152df)

**Comment by Ankush263** [comment]
You can use this
`value: parseEther(Amount) as never`
this probably solves the issue.

**Comment by tmm** [comment, marked as answer]
Fixed in [viem@1.0.1](https://github.com/wagmi-dev/viem/releases/tag/viem%401.0.1). Just to upgrade to that version.

**Reply by jcislo** [reply]
I wish it were that easy... this breaks all of my dependancies... I currently have rainbowkit 1.0.1 and wagmi 1.0.1

**Reply by Ankush263** [reply]
does it works `value: parseEther(Amount) as never` ?? on older version...

**Reply by jcislo** [reply]
@Ankush263 I first tried "as never" before changing packages and it did not work. What fixed it however was upgrading viem to 0.3.5 (not 1.0.1 like stated above). I also ran the highest version of wagmi possible (according to npm in my console.)

Unfortunately I had to run npm using "--force" to make this work otherwise the dependancies won't match. RainbowKit has not been updated to match the newest wagmi and viem I don't think.
